src/adaptors/hop-protocol/index.js

```diff
@@ -0,0 +1,115 @@
+const axios = require('axios');
+const sdk = require('@defillama/sdk');
+const superagent = require('superagent');
+const { default: BigNumber } = require('bignumber.js');
+
+const getCoreConfig = async () => {
+  const {
+    data: { bridges },
+  } = await axios.get(
+    'https://assets.hop.exchange/mainnet/v1-core-config.json'
+  );
+
+  return bridges;
+};
+
+const getPoolTokenBalances = async (coreConfig, chain, token) => {
+  const config = coreConfig[token][chain];
+  const poolAddress = config.l2SaddleSwap;
+  const tokenAddress = config.l2CanonicalToken;
+  const hopTokenAddress = config.l2HopBridgeToken;
+  const adaptedChain = getAdaptedChain(chain);
+
+  const [tokenBalance, hopTokenBalance, decimals] = await Promise.all(
+    [
+      sdk.api.abi.call({
+        abi: 'erc20:balanceOf',
+        chain: adaptedChain,
+        target: tokenAddress,
+        params: [poolAddress],
+      }),
+      sdk.api.abi.call({
+        abi: 'erc20:balanceOf',
+        chain: adaptedChain,
+        target: hopTokenAddress,
+        params: [poolAddress],
+      }),
+      sdk.api.abi.call({
+        abi: 'erc20:decimals',
+        chain: adaptedChain,
+        target: tokenAddress,
+        params: [],
+      }),
+    ].map((promise) => promise.then((data) => data.output))
+  );
+
+  return { tokenBalance, hopTokenBalance, decimals, tokenAddress };
+};
+
+const getPoolTvl = async (coreConfig, chain, token) => {
+  const { tokenBalance, hopTokenBalance, decimals, tokenAddress } =
+    await getPoolTokenBalances(coreConfig, chain, token);
+  const key = `${chain}:${tokenAddress}`;
```

Comment:
make sure to pass the correct chain value for gnosis, in our case `xdai`, I'd just do:
```
  const key =
    chain === 'gnosis' ? `xdai:${tokenAddress}` : `${chain}:${tokenAddress}`;
```